internal/host/hostcommands/install_cmd.go

```diff
@@ -325,6 +335,45 @@ func constructHostInstallerArgs(cluster *common.Cluster, host *models.Host, inve
 	return toJSONString(installerArgs)
 }
 
+func appendISCSIArgs(installerArgs []string, disk *models.Disk, inventory *models.Inventory, hasUserConfiguredIP bool) ([]string, error) {
+	// enable iSCSI on boot
+	installerArgs = append(installerArgs, "--append-karg", "rd.iscsi.firmware=1")
+
+	if hasUserConfiguredIP {
+		return installerArgs, nil
+	}
+
+	// configure DHCP on the interface used by the iSCSI boot volume
+	iSCSIHostIP := net.ParseIP(disk.Iscsi.HostIPAddress)
+	if iSCSIHostIP == nil {
+		return nil, fmt.Errorf("Cannot parse iSCSI host IP %s", disk.Iscsi.HostIPAddress)
+	}
+	isIPv6 := govalidator.IsIPv6(iSCSIHostIP.String())
+
+	nic := funk.Find(inventory.Interfaces, func(nic *models.Interface) bool {
+		ips := nic.IPV4Addresses
+		if isIPv6 {
+			ips = nic.IPV6Addresses
+		}
+		return funk.Contains(ips, func(ip string) bool {
+			nicIP, _, _ := net.ParseCIDR(ip)
+			return nicIP.String() == iSCSIHostIP.String()
+		})
+	}).(*models.Interface)
+
+	if nic == nil {
+		return nil, fmt.Errorf("Cannot find the interface belonging to iSCSI host IP %s", iSCSIHostIP.String())
+	}
+
+	dhcp := "dhcp"
+	if isIPv6 {
+		dhcp = "dhcp6"
+	}
+	installerArgs = append(installerArgs, "--append-karg", fmt.Sprintf("ip=%s:%s", nic.Name, dhcp))
```

Comment:
not sure, but please validate we don't append multiple statements of "--append-karg"
lines 340, 321, 372

Reply:
we will append multiple statements depending on the needs. I performed a cluster installation to check that.